### ISSUE TITLE
Run `cargo miri test` in CI for unsafe crates

### ISSUE THREAD

**Issue by Manishearth** [unicode-org/icu4x#1259]
We probably should run `cargo miri test` in CI for `zerovec` and `yoke`.  This has to be done with a nightly toolchain.

This will help catch issues in the unsafe code.

**Comment by Fenz-art** [comment]
Hi! I'd like to work on this issue. I noticed that #6723 reports existing Miri failures in zerovec. Should #1259 wait until those are resolved, or would you still like the CI job added (perhaps as an allowed failure)?

**Comment by Manishearth** [comment]
Yes, something that marks it as an allowed failure would be good, and we can link to the existing issue.

**Comment by Fenz-art** [comment]
Thanks for the guidance!

I added a non-blocking Miri CI job for `zerovec` and `yoke` as suggested.

Changes:
- Added a new `ci-job-miri` task in `Makefile.toml`
- Added it to the `ci-all` aggregate task
- Added a new `miri` GitHub Actions job with `continue-on-error: true`
- The job uses ICU4X's pinned nightly toolchain and runs:
  `cargo +${PINNED_CI_NIGHTLY} miri test -p zerovec --features serde -p yoke`

I verified locally that:
- `cargo make ci-job-miri` starts correctly
- Miri setup works
- The intended zerovec/yoke Miri tests execute successfully

I'll open a PR shortly and link it here.

**Comment by Fenz-art** [comment]
I've opened PR #8222 implementing the non-blocking Miri CI job following the discussion above. Thanks for the guidance!

**Comment by Fenz-art** [comment]
``` The first CI run exposed an issue where PINNED_CI_NIGHTLY isn't available in the GitHub Actions environment for pull_request events, so the Miri installation step receives an empty toolchain name. I'm investigating the existing nightly workflow pattern and will update the PR accordingly. ```
